rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java

```diff
@@ -953,5 +955,72 @@ public Integer loadPreferredClearingDateLimit() {
         // returning default value 7 (days) if variable is not set
         return limit < 1 ? 7 : limit;
     }
-}
+
+
+    public List<LicenseInfoParsingResult> processLicenseInfoWithObligations(
+            List<LicenseInfoParsingResult> licenseInfoWithObligations, Map<String, String> releaseIdToAcceptedCLI,
+            List<Release> releases, User user) throws TException {
+        ThriftClients thriftClients = new ThriftClients();
+        LicenseInfoService.Iface licenseClient = thriftClients.makeLicenseInfoClient();
+
+        for (Release release : releases) {
+            List<Attachment> approvedCliAttachments = SW360Utils.getApprovedClxAttachmentForRelease(release);
+            if (approvedCliAttachments.isEmpty()) {
+                log.info("No approved CLX attachments found for release: {}. Proceeding with attached CLX.",
+                        release.getId());
+                approvedCliAttachments = SW360Utils.getClxAttachmentForRelease(release);
+            }
+            final String releaseId = release.getId();
+
+            if (approvedCliAttachments.size() == 1) {
```

Comment:
What will happen if there are more than 1 approved CLI attached? What is expected here?

Reply:
done.